pkg/utils/retry.go

```diff
@@ -48,18 +48,14 @@ func Retry(interval time.Duration, maxRetries int, f ConditionFunc) error {
 
 	for i := 0; ; i++ {
 		ok, err := f()
-		if err != nil {
+		if err != nil && i == maxRetries {
```

Comment:
Returns the error when the max retries number has reached.

Reply:
The `RetryError` type is no longer used after this changed.
Should it be kept or removed as well?

Reply:
Good point, I reckon that I could remove it if it is no longer used.